src/main/java/org/commonjava/util/partyline/impl/infinispan/model/InfinispanJFS.java

```diff
@@ -1,659 +1,132 @@
-            {
-                logger.trace( "INIT: locking directory WITHOUT lock in underlying filesystem!" );
-                output = null;
-                //                fileLock = null;
-                joinable = false;
-            }
-            else if ( doOutput )
-            {
-                logger.trace( "INIT: read-write JoinableFile: {}", target );
-                output = new JoinableOutputStream();
-            }
-            else
-            {
-                logger.trace( "INIT: read-only JoinableFile: {}", target );
-                output = null;
-                logger.trace( "INIT: set flushed length to: {}", target.length() );
-                flushed.set( target.length() );
-            }
-        }
-        catch ( OverlappingFileLockException e )
-        {
-            throw new IOException( "Cannot lock file: " + target + ". Reason: " + e.getMessage() + "\nLocked by: "
-                                           + owner.getLockInfo(), e );
-        }
-    }
+    private String nodeKey;
 
-    public LocalLockOwner getLockOwner()
-    {
-        return owner;
-    }
+    private final Cache<String, FileMeta> metadataCache;
 
-    // only public for testing purposes...
-    public OutputStream getOutputStream()
-    {
-        return output;
-    }
+    private final Cache<UUID, FileBlock> blockCache;
 
-    public boolean isJoinable()
+    public InfinispanJFS( final String nodeKey, final Cache<String, FileMeta> metadataCache,
+                          final Cache<UUID, FileBlock> blockCache )
     {
-        return joinable;
+        this.nodeKey = nodeKey;
+        this.metadataCache = metadataCache;
+        this.blockCache = blockCache;
     }
 
     @Override
-    public boolean isDirectory()
+    public JoinableFile getFile( final File file, final LocalLockOwner lockOwner, final StreamCallbacks callbacks,
+                                 final boolean doOutput, final ReentrantOperationLock opLock ) throws IOException
     {
-        return metadata.isDirectory();
-    }
-
-    public InputStream joinStream()
-            throws IOException, InterruptedException
-    {
-        return lockAnd( (lock)->{
-            if ( !joinable )
-            {
-                // if the channel is null, this is a directory lock.
-                throw new IOException( "JoinableFile is not accepting join() operations" );
-            }
-
-            JoinInputStream result = new JoinInputStream( inputs.size() );
-            inputs.put( result.hashCode(), result );
-
-            Logger logger = LoggerFactory.getLogger( getClass() );
-            logger.debug( "JOIN: {} (new joint count: {})", Thread.currentThread().getName(), inputs.size() );
-
-            return result;
-        });
+        return new InfinispanJF( file, lockOwner, callbacks, doOutput, opLock, this );
     }
 
     @Override
-    public boolean isWriteLocked()
+    public LocalLockManager getLocalLockManager()
```

Comment:
We should make sure this is consistent with the RandomAccessFile form. I'm not sure where these LocalLockManager instances are created.